src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -61,5 +73,77 @@ internal void RemovePredecessor(BasicBlock block)
         {
             _predecessors.Remove(block);
         }
+
+        [Flags]
+        public enum BranchFlags
+        {
+            Regular = 0x01,
+
+            /// <summary>
+            /// This is the only flag that can be combined with other values.
+            /// </summary>
+            Error = 0x02,
+
+            Return = 0x04,
+            StructuredExceptionHandling = 0x08,
+            ProgramTermination = 0x10,
```

Comment:
>ProgramTermination [](start = 12, length = 18)

I saw "(Regular)" and "(StructuredExceptionHandling)" in the tests, but couldn't find any instance of "(Return)", "(ProgramTermination)" or "(Throw)". Is that expected?

Reply:
> I saw "(Regular)" and "(StructuredExceptionHandling)" in the tests, but couldn't find any instance of "(Return)", "(ProgramTermination)" or "(Throw)". Is that expected?

Yes, this is expected, those are not handled yet.

---
In reply to: [167950767](https://github.com/dotnet/roslyn/pull/24770#discussion_r167950767) [](ancestors = 167950767)